vmm/src/vm.rs

```diff
@@ -417,12 +418,13 @@ impl DeviceManager {
         virtio_device: Box<vm_virtio::VirtioDevice>,
         memory: GuestMemoryMmap,
         allocator: &mut SystemAllocator,
-        vm_fd: &VmFd,
+        vm_fd: &Arc<VmFd>,
         pci_root: &mut PciRoot,
         mmio_bus: &mut devices::Bus,
     ) -> DeviceManagerResult<()> {
-        let mut virtio_pci_device = VirtioPciDevice::new(memory, virtio_device)
-            .map_err(DeviceManagerError::VirtioDevice)?;
+        let mut virtio_pci_device =
+            VirtioPciDevice::new(memory, virtio_device, DEFAULT_MSIX_VEC_NUM)
```

Comment:
DEFAULT_MSIX_VEC_NUM  is 2, how can we guarantee this is enough for all the supported virtio device types. I mean in future when we add a device that requires more than 2 vectors it's easy for us to forget to change here.

Reply:
Well for now I'm trying to match the behavior we had with pin based irq, and the virtio spec specifies that 2 vectors is the minimum you must specify for a virtio device.
But in the future, we should consider tweaking this number based on the number of queues we use from our virtio devices if we want better performances.